cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/SparkProgramRuntimeProvider.java

```diff
@@ -433,6 +433,11 @@ public URL getResource(String name) {
       }
       // resource = jar:file:/path/to/cdap/lib/org.scala-lang.scala-library-2.10.4.jar!/library.properties
       // baseClasspath = /path/to/cdap/lib/org.scala-lang.scala-library-2.10.4.jar
+
+      // ignoring jrt scheme for java11 support
+      if (resource.getProtocol().equals("jrt")) {
```

Comment:
The root of the problem is inside the `ClassLoaders.getClassPathURL` method, we should fix it there instead of here. That util class is used in multiple places, it is better to fix it once.

Reply:
> The root of the problem is inside the `ClassLoaders.getClassPathURL` method, we should fix it there instead of here. That util class is used in multiple places, it is better to fix it once.

I thought about this. But `ClassLoaders.getClassPathURL` does not return null today. It might break in lot of other places including this method.